src/org/opendatakit/briefcase/export/Model.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.opendatakit.briefcase.export;
+
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static org.javarosa.core.model.Constants.DATATYPE_NULL;
+import static org.opendatakit.briefcase.export.DataType.GEOPOINT;
+import static org.opendatakit.briefcase.export.DataType.NULL;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Stream;
+import org.javarosa.core.model.instance.TreeElement;
+
+/**
+ * This class represents a particular level in the model of a Form.
+ * It can hold the root level model or any of its fields.
+ */
+class Model {
+  private final TreeElement model;
+
+  /**
+   * Main constructor for {@link Model} that takes a {@link TreeElement} as its root.
+   */
+  Model(TreeElement model) {
+    this.model = model;
+  }
+
+  /**
+   * Method that lets iterate and (flat)map over all the children {@link Model} members of
```

Comment:
Hey, let’s iterate! :-)
I suggest,
~~Method that lets~~ Iterates and (flat)maps over ...